content/blog/2023/2023-03-13-apollo-graphql-nodejs-server.md

```diff
@@ -0,0 +1,653 @@
+
+Now, we should be able to edit students details inline, we can use GraphQL Playground to do this:
+
+{{% image alt="Graphql update student" src="images/posts/node-graphql/update-student.png" %}}
+
+### Delete Student Details API
+
+Lastly, we are going to attempt deleting students from our database. Delete mutation is similar to create and update mutation from the previous section. We simply require a mutation that takes the id of the student data to be deleted.
+
+To add the delete feature to our application update the schema in `models/typeDefs.js` by adding `delete` mutation that takes `id` argument and returns the student object if successful:
+
+```javascript
+const gql = require("graphql-tag");
+
+const typeDefs = gql`
+  type Query {
+    hello: String
+    welcome(name: String): String
+    students: [Student] #return array of students
+    student(id: ID): Student #return student by id
+  }
+  type Student {
+    id: ID
+    firstName: String
+    lastName: String
+    age: Int
+  }
+  type Mutation {
+    create(firstName: String, lastName: String, age: Int): Student
+    update(id: ID, firstName: String, lastName: String, age: Int): Student
+    delete(id: ID): Student
+  }
+`;
+
+module.exports = { typeDefs };
+```
+
+Update the `resolvers.js` file to implement the `delete` resolver function:
+
+```javascript
+const { Student } = require("./models/Student.js");
+
+// GraphQL Resolvers
+const resolvers = {
+  Query: {
+    hello: () => "Hello from Reflectoring Blog",
+    welcome: (parent, args) => `Hello ${args.name}`,
+    students: async () => await Student.find({}), // return array of students
+    student: async (parent, args) => await Student.findById(args.id), // return student by id
+  },
+  Mutation: {
+    create: async (parent, args) => {
+      const { firstName, lastName, age } = args;
+      const newStudent = new Student({
+        firstName,
+        lastName,
+        age,
+      });
+      await newStudent.save();
+      return newStudent;
+    },
+    update: async (parent, args) => {
+      const { id } = args;
+      const updatedStudent = await Student.findByIdAndUpdate(id, args);
+      if (!updatedStudent) {
+        throw new Error(`Student with ID ${id} not found`);
+      }
+      return updatedStudent;
+    },
+    delete: async (parent, args) => {
+      const { id } = args;
+      const deletedStudent = await Student.findByIdAndDelete(id);
+      if (!deletedStudent) {
+        throw new Error(`Student with ID ${id} not found`);
+      }
+      return deletedStudent;
+    },
+  },
+};
+
+module.exports = { resolvers };
+```
+
+Grab a student id from the database, then delete the student in the GraphQL playground:
+
+{{% image alt="Graphql delete student" src="images/posts/node-graphql/delete-student.png" %}}
+
+Great news! Our CRUD APIs on the backend are now operational!
+
+To ensure that everything is working properly, double-check that students are being created, deleted, and updated in the database.
```

Comment:
Given that you have provided instructions on how to install and set up MongoDb, would you be able to include a section detailing how to save the model to the database? This would provide readers with a comprehensive understanding of the use case, from the client call to the actual data storage in the database.

Reply:
Please I think I need more clarification on what you mean here, and also I was avoiding the article being too long that why I didn't delve into most sections.